packages/pluggableWidgets/accordion-web/src/utils/AccordionGroupStateReducer.ts

```diff
@@ -0,0 +1,41 @@
+import { AccGroup } from "../components/AccordionGroup";
+
+export type AccordionGroupsReducerAction =
+    | { type: "sync"; groups: AccGroup[] }
+    | { type: "expand" | "collapse"; group: AccGroup };
+
+export function getAccordionGroupsReducer(
+    expand: "single" | "multiple" | "all"
+): (state: AccGroup[], action: AccordionGroupsReducerAction) => AccGroup[] {
+    return (state: AccGroup[], action: AccordionGroupsReducerAction): AccGroup[] => {
+        if (action.type === "sync" || expand === "single") {
+            const newState = action.type === "sync" ? action.groups : state;
+
+            return newState.map((group, index) => {
+                let collapsed;
+
+                if (action.type === "sync") {
+                    collapsed = state[index].collapsed; // The previous collapsed state of the group at the same index can be used, because the order of groups remains the same throughout the lifetime of the widget.
```

Comment:
- So if I'm correct, the `sync` action just syncs in the incoming `action.groups` AccGroup items but keeps its previous `collapsed` state?
- If so, why is the behaviour not that the new group just overrides everything?
- Can this widget be used with dynamic values? If so, does the comment still hold?

Reply:
- Correct.
- You don't want to override everything, because that would reinit the collapsed state on every data change.
- The ReactNode get updated when there is a change in the contained widgets. Also the visibility prop is dynamic. Therefore, this comment doesn't hold.